sdk/src/client/bindings/python/iota_client/client.py

```diff
@@ -382,4 +526,334 @@ def as_dict(self):
 
 class CoinType(Enum):
     IOTA = 4218
-    SHIMMER = 4219
\ No newline at end of file
+    SHIMMER = 4219
+
+class UnlockConditionType(Enum):
+    Address = 0
+    StorageDepositReturn = 1
+    Timelock = 2
+    Expiration = 3
+    StateControllerAddress = 4
+    GovernorAddress = 5
+    ImmutableAliasAddress = 6
+
+class UnlockCondition():
+    def __init__(self, type=None, address=None, amount=None, unix_time=None, return_address=None) -> None:
+        """Initialize an UnlockCondition
+        
+        Parameters
+        ----------
+        type : UnlockConditionType
+            The type of unlock condition
+        address : Address
+            Address for unlock condition
+        amount : int
+            Amount for storage deposit unlock condition
+        unix_time : int
+            Unix timestamp for timelock and expiration unlock condition
+        return_address : Address
+            Return address for expiration and storage deposit unlock condition
+        """
+        self.type = type
+        self.address = address
+        self.amount = amount
+        self.unix_time = unix_time
+        self.return_address = return_address
+
+    def as_dict(self):
+        config = {k: v for k, v in self.__dict__.items() if v != None}
+        
+        if 'type' in config:
+            config['type'] = config['type'].value
+
+        if 'address' in config:
+            config['address'] = config['address'].as_dict()
+
+        if 'return_address' in config:
+            config['return_address'] = config['return_address'].as_dict()
+
+        if 'amount' in config:
+            config['amount'] = str(config['amount'])
+
+        return config
+
+class AddressUnlockCondition(UnlockCondition):
+    def __init__(self, address):
+        """Initialize an AddressUnlockCondition
+        
+        Parameters
+        ----------
+        address : Address
+            Address
+        """
+        super().__init__(type=UnlockConditionType.Address, address=address)
+
+class StorageDepositReturnUnlockCondition(UnlockCondition):
+    def __init__(self, amount, return_address):
+        """Initialize a StorageDepositReturnUnlockCondition
+        
+        Parameters
+        ----------
+        amount : int
+            Amount
+        return_address : Address
+            Return address
+        """
+        super().__init__(type=UnlockConditionType.StorageDepositReturn, amount=amount, return_address=return_address)
```

Comment:
Just curious, when they have the proper name already, are named field necessary?

Reply:
Not afaik